release-notes/7.0/preview/api-diff/preview4/Microsoft.NETCore.App/7.0-preview4_System.Runtime.InteropServices.md

```diff
@@ -0,0 +1,248 @@
++        public struct GenericPlaceholder
++    }
++    public enum CustomTypeMarshallerDirection {
++        In = 1,
++        None = 0,
++        Out = 2,
++        Ref = 3,
++    }
++    public enum CustomTypeMarshallerFeatures {
++        CallerAllocatedBuffer = 2,
++        None = 0,
++        TwoStageMarshalling = 4,
++        UnmanagedResources = 1,
++    }
++    public enum CustomTypeMarshallerKind {
++        LinearCollection = 1,
++        Value = 0,
++    }
++    public sealed class MarshalUsingAttribute : Attribute {
++        public const string ReturnsCountValue = "return-value";
++        public MarshalUsingAttribute();
++        public MarshalUsingAttribute(Type nativeType);
++        public int ConstantElementCount { get; set; }
++        public string CountElementName { get; set; }
++        public int ElementIndirectionDepth { get; set; }
++        public Type? NativeType { get; }
++    }
++    public sealed class NativeMarshallingAttribute : Attribute {
++        public NativeMarshallingAttribute(Type nativeType);
++        public Type NativeType { get; }
++    }
+-    public readonly struct NFloat : IComparable, IComparable<NFloat>, IEquatable<NFloat>, IFormattable, ISpanFormattable {
++    public readonly struct NFloat : IAdditionOperators<NFloat, NFloat, NFloat>, IAdditiveIdentity<NFloat, NFloat>, IBinaryFloatingPointIeee754<NFloat>, IBinaryNumber<NFloat>, IBitwiseOperators<NFloat, NFloat, NFloat>, IComparable, IComparable<NFloat>, IComparisonOperators<NFloat, NFloat>, IDecrementOperators<NFloat>, IDivisionOperators<NFloat, NFloat, NFloat>, IEqualityOperators<NFloat, NFloat>, IEquatable<NFloat>, IExponentialFunctions<NFloat>, IFloatingPoint<NFloat>, IFloatingPointIeee754<NFloat>, IFormattable, IHyperbolicFunctions<NFloat>, IIncrementOperators<NFloat>, ILogarithmicFunctions<NFloat>, IMinMaxValue<NFloat>, IModulusOperators<NFloat, NFloat, NFloat>, IMultiplicativeIdentity<NFloat, NFloat>, IMultiplyOperators<NFloat, NFloat, NFloat>, INumber<NFloat>, INumberBase<NFloat>, IParsable<NFloat>, IPowerFunctions<NFloat>, IRootFunctions<NFloat>, ISignedNumber<NFloat>, ISpanFormattable, ISpanParsable<NFloat>, ISubtractionOperators<NFloat, NFloat, NFloat>, ITrigonometricFunctions<NFloat>, IUnaryNegationOperators<NFloat, NFloat>, IUnaryPlusOperators<NFloat, NFloat> {
++        static NFloat E { get; }
+-        public static NFloat Epsilon { get; }
++        static NFloat Epsilon { get; }
+-        public static NFloat MaxValue { get; }
++        static NFloat MaxValue { get; }
+-        public static NFloat MinValue { get; }
++        static NFloat MinValue { get; }
+-        public static NFloat NaN { get; }
++        static NFloat NaN { get; }
+-        public static NFloat NegativeInfinity { get; }
++        static NFloat NegativeInfinity { get; }
++        static NFloat NegativeZero { get; }
++        static NFloat Pi { get; }
+-        public static NFloat PositiveInfinity { get; }
++        static NFloat PositiveInfinity { get; }
```

Comment:
The diff is stripping the `public` prefix. Likely doesn't understand when a type member implements a `static abstracts in interfaces member`
```suggestion
+        static NFloat E { get; }
+        static NFloat NegativeZero { get; }
+        static NFloat Pi { get; }
```

Reply:
The provided suggestion is confusing compared to the description of the problem (the suggestion seems to be removed a bunch of properties and removing 3 public keywords).

Can you please look at this file again? The `public` keyword is back, and the APIs were preserved. Is that what you expected?

Reply:
> The provided suggestion is confusing compared to the description of the problem (the suggestion seems to be removed a bunch of properties and removing 3 public keywords).

The properties no longer listed had no changes and so shouldn't have been in the diff at all. Ultimately we added 3 new `public static` properties (notably I missed the public keyword in my suggestion)

Reply:
The updated version of the file you pushed looks correct now.